bundle/config/validate/required.go

```diff
@@ -0,0 +1,84 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sort"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/internal/validation/generated"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+)
+
+type required struct{}
+
+func Required() bundle.Mutator {
+	return &required{}
+}
+
+func (f *required) Name() string {
+	return "validate:required"
+}
+
+func (f *required) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+
+	// Generate prefix tree for all required fields.
+	trie := &dyn.TrieNode{}
+	for k := range generated.RequiredFields {
+		pattern, err := dyn.NewPatternFromString(k)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("invalid pattern %q for required field validation: %w", k, err))
+		}
+
+		err = trie.Insert(pattern)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("failed to insert pattern %q into trie: %w", k, err))
+		}
+	}
+
+	err := dyn.WalkReadOnly(b.Config.Value(), func(p dyn.Path, v dyn.Value) error {
+		// If the path is not found in the prefix tree, we do not need to validate any required
+		// fields in it.
+		pattern, ok := trie.SearchPath(p)
+		if !ok {
+			return nil
+		}
+
+		cloneP := slices.Clone(p)
+
+		fields := generated.RequiredFields[pattern.String()]
+		for _, field := range fields {
+			vv := v.Get(field)
+			if vv.Kind() == dyn.KindInvalid || vv.Kind() == dyn.KindNil {
+				diags = diags.Append(diag.Diagnostic{
```

Comment:
I think it can be converted to use `logdiag.LogDiag`

Reply:
If I understood the idea behind `logdiag` is that eventually we want to replace all mutator to never return anything. Can you instead sort `fields` to make it determenistic?